### ISSUE TITLE
Help troubleshooting MAMEnrollmentStatusInternalError

### ISSUE THREAD

**Issue by miguelweiss** [msintuneappsdk/intune-app-sdk-xamarin#51]
The issue is related with the registering and enrolling an account process with an iOS application.  

We are working on an iOS Xamarin based application for **iOS 15 iPhone**, Building with **Xcode 12.5.1**, using **MSAL** authentication and using the **Microsoft.Intune.MAM.Xamarin.iOS 15.0.0** Nuget package.

When we call the `IntuneMAMEnrollmentManager.Instance.RegisterAndEnrollAccount(string identity)` for registering an account, then in the `public virtual void EnrollmentRequestWithStatus (IntuneMAMEnrollmentStatus status)` method we are capturing an **IntuneMAMEnrollmentStatus** object with the **error message** = _There was an internal error in the Intune MAM SDK.  Please see the NSError object for more details._, a **NSError** = _null_ and a **StatusCode** = _MAMEnrollmentStatusInternalError_.
Check the below image for details:
![image](https://user-images.githubusercontent.com/83839889/134551197-128a1af1-a1ba-450c-9096-4c32efe44170.png)

In the **Entitlements.plist** file we have the following:
![image](https://user-images.githubusercontent.com/83839889/134552481-f349d0a8-8cdb-419c-9e6a-140a26cbf3a9.png)

The interesting thing is that we do not get an error message and are able to register and enroll the user account when we use the **Microsoft.Intune.MAM.Xamarin.iOS 14.3.5** Nuget package.

Any idea how we can troubleshoot this issue or have any clue on what is the problem?

Thank you very much.

**Comment by darosal** [comment]
Hi @miguelweiss, as part of this release we have also switched over from ADAL to MSAL. Have you updated your Info.plist to reflect an MSAL redirect URI? You can follow steps here [Configure MSAL](https://docs.microsoft.com/en-us/mem/intune/developer/app-sdk-ios#configure-msal) (ignore any parts that aren't relevant to Xamarin). If the issue persists after making these changes please let me know!

**Comment by Aculeo** [comment]
I'm facing this issue with the iOS package **15.1.3**. Also after downgrading this error went away. I'm on Xamarin.Forms 5.0.0.2012. The app is using MSAL.

**Comment by darosal** [comment]
@Aculeo, can you confirm if you have PackageReference set in your csproj file? See #52.

**Comment by Aculeo** [comment]
Yep, I'm using package references already (this is the reference of a package version that's working).

```
<PackageReference Include="Microsoft.Intune.MAM.Xamarin.iOS">
    <Version>14.1.3</Version>
</PackageReference>
```

**Comment by Aculeo** [comment]
Hi,
Now the error has changed to this with the latest version:

```
MAMEnrollmentStatusADALInternalError: The operation failed because the SDK encountered an internal error in ADAL when attempting to acquire the user's MAM token.
```

**Comment by MikeP0911** [comment]
Hello!
We face the same issue.
Our application does not use ADAL or MSAL.
Till Intune 14.3.5 there is no error.
But there is no error if Updated Intune version is used.
But the Intune login is also not happening.

Any help is appreciated.
Thank you

**Comment by MikeP0911** [comment]
@darosal 
Our application lets the Intune SDK to handle the authentication.
Our application was designed using this link. (https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-xamarin)
Now when Intune Login button is clicked, a blank page is displayed.
We couldn't find any logs.
If the Intune SDK 14.3.5 is used the application works.
But once updated to next Intune SDK version, it does not work.
Our application does not use ADAL and MSAL.
So we are confused of what to be implemented.

Help is much appreciated.
Thank you

**Comment by aaulmer** [comment]
We tested integration to our app using the latest version of 17.1.1 (released December 21, 2022) and we see the exact same issue. Our app uses MSAL, and we are passing the identity from MSAL after authenticating the user account. Is there a stable version after 14.3.5 that we can rollback to until this is fixed?

**Comment by MikeP0911** [comment]
@aaulmer 14.3.5 is the last stable version as far I tested. My application does not use MSAL. we use the Intune SDK authentication. But still we face the same issue.

**Comment by Kyle-Reis** [comment]
Hi @MikeP0911 and @aaulmer - I am following up with each of you separately, as I believe the issues you are each facing could be slightly different, and don't want to use this previously-existing to track both of them.

**Comment by gustavoja** [comment]
Hello I'm having the same issue on an Xamarin iOS native app following these steps:

1. Login successfuly with MSAL first to receive an access token
2. Try to enrol with Intune using IntuneMAMEnrollmentManager.Instance.RegisterAndEnrollAccount(upn)
3. On the IntuneMAMEnrollmentDelegate I get an error "There was an internal error in the Intune MAM SDK. Please see the NSError object for more details." The NSerror is null.

![Screenshot 2023-01-29 at 23 19 16](https://user-images.githubusercontent.com/3488092/215358808-2f1e4d16-b801-4086-8658-68fed818997b.png)

I configured the info.plist section with the values for 
ADALRedirectUr : msauth.xxx.xxxx.xxxxx://auth
ADALClientId  : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
ADALAuthority : https://login.microsoftonline.com/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx 
Tried all kind of combinations with no success. Always the same error.

I also tried overriding the values on runtime, same result.

IntuneMAMSettings.AadRedirectUriOverride : msauth.xxx.xxxx.xxxxx://auth
IntuneMAMSettings.AadClientIdOverride xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
IntuneMAMSettings.AadAuthorityUriOverride https://login.microsoftonline.com/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx


On the BE side i gave the permission to the MSAL app to access DeviceManagementManagedApps.ReadWrite. The scope of the MSAL token received is configured to request that same value.
https://wip.mam.manage.microsoft.us//DeviceManagementManagedApps.ReadWrite

I'm using
- Microsoft.Intune.MAM.Xamarin.iOS 17.1.1
- Microsoft.Identity.Client 4.49.1

**Without MSAL**

If I remove  the MSAL values on the Info.plist and then try to login/enroll directly with Intune calling:
IntuneMAMEnrollmentManager.Instance.LoginAndEnrollAccount(upn)
The enrollment is completed successfuly and I receive the security policies on my device

![Screenshot 2023-01-29 at 23 05 30](https://user-images.githubusercontent.com/3488092/215358242-26f05907-9518-474d-8533-7f305fdb708b.png)

Also putting VerboseLoggingEnabled or ADALLogOverrideDisabled to true/false doesn't change anything on the log of the app.

Can anyone help me out here?
Thank you very much!

**Comment by Kyle-Reis** [comment]
Hi @gustavoja , please file a separate issue and I will follow up with you on it. Thanks.

**Comment by gustavoja** [comment]
Hello @Kyle-Reis I created the new issue here https://github.com/msintuneappsdk/intune-app-sdk-xamarin/issues/57
Hope you can help me with it
Thanks
